Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -187,30 +187,55 @@ class GameController : public Core::Device
   };
 
   // Rumble
-  template <int LowEnable, int HighEnable, int SuffixIndex>
-  class GenericMotor : public Output
+  class Motor : public Output
   {
   public:
-    explicit GenericMotor(SDL_GameController* gc) : m_gc(gc) {}
-    std::string GetName() const override
+    Motor(const char* name, GameController& gc, Uint16* state)
+        : m_name{name}, m_gc{gc}, m_state{*state}
     {
-      return std::string("Motor") + motor_suffixes[SuffixIndex];
     }
+    std::string GetName() const override { return m_name; }
```

Comment:
Any reason this makes a copy?

Reply:
The virtual function of the base class `Output` returns `std::string`. This has to match.